GovUk.Frontend.ExampleApp/Views/NotificationBanner/Index.cshtml

```diff
@@ -16,6 +16,30 @@
     <p class="govuk-body">Do not use a notification banner to tell users that they’ve finished using a linear service. Use a <a class="govuk-notification-banner__link" href="https://design-system.service.gov.uk/patterns/confirmation-pages/">confirmation page</a> instead.</p>
 </govuk-notification-banner>
 
+<tpr-notification-banner>
+    <p class="tpr-notification-banner__heading">
+       TPR Notification Banner
+    </p>
+    <p class="govuk-body">The TPR Notification Banner lets you tell the user something that's not directly relevant to the thing they're trying to do on that page.</p>
+    <p class="govuk-body">Buttons or button groups can be added. Below is an example of a button group.</p>
+    <div class="govuk-button-group">
+        <govuk-button >Default button</govuk-button>
+        <govuk-button class="govuk-button--secondary" type="button">Secondary button</govuk-button>
+        <govuk-button class="govuk-button--warning" type="button">Warning button</govuk-button>
+        <a href="/" class="govuk-link">Cancel</a>
+    </div>
+</tpr-notification-banner>
+
+<p class="govuk-body">Below is an example of how the TPR Notification Banner might be used with a button to dismiss the banner as part of a page.</p>
+
+<tpr-notification-banner>
+    <p class="tpr-notification-banner__heading">
+        You will be logged out in 30 seconds due to inactivity
+    </p>
+    <p class="govuk-body">Click the button below to dismiss the message and continue.</p>
+    <govuk-button class="govuk-button--secondary">Dismiss</govuk-button>
```

Comment:
There is no guidance here about whether this would be a JS-only feature. If it is then you would want to render the Umbraco-configurable details as `data-*` attributes and then have JS that picked up on those and turned them into a button.

Reply:
I've added guidance about this being a JS only feature, however I don't think I understand what you mean by having the Umbraco-configurable details as `data-*`. Can you explain more or is there an example I can follow?

Reply:
Now that you've made your example a bit more focussed I don't think you need to use any `data-*` attributes. But your example should degrade gracefully in a no-JS scenario as people will be following your example. For a dismiss button you can do that by assigning it a class of `govuk-!-display-none` so that it doesn't display for a no-JS scenario, then have JS remove that class so that it does display when JS is available. It would also be good to have the second Dismiss button similarly wired up.

Reply:
Both buttons are hidden when JS is unavailable and when JS is available they both dismiss their banner